### PR TITLE
Fix: text width calculation

### PR DESCRIPTION
**Pull request by mlange-42**
Use the text length in runes, not bytes.

### CODE REVIEW

File: render/render.go [hunk 1]
```diff
@@ -178,7 +178,7 @@ func fitText(text string, fontSize int, W float64) (scale float64, h float64) {
 }
 
 func textWidth(text string, fontSize float64) float64 {
-	return fontSize * float64(len(text)) * textWidthMultiplier
+	return fontSize * float64(len([]rune(text))) * textWidthMultiplier
```
Comment (nikolaydubina) [inline render/render.go:181]: nice

could you use this instead? https://pkg.go.dev/unicode/utf8#RuneCountInString

**Review by nikolaydubina — CHANGES_REQUESTED** [review]

File: render/render.go (hunk 1 above)
Comment (mlange-42) [reply render/render.go:181]: Cool, thanks! Was not aware of that, also already asking myself if it isn't a waste of resources that string length needs to be calculates like this in Go. Will do.

Comment (mlange-42) [reply render/render.go]: Done.

Comment (nikolaydubina) [reply render/render.go]: that's UTF8, so it is O(N). good to have some optimisation! tbh I see lots of non-obviously complex things are done really well by Go core (other example is integer parsing/printing). anws lgtm!

File: render/render_test.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		{
+			text:     "♠♧♡",
+			expWidth: 3.0,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.text, func(t *testing.T) {
+			w := textWidth(tc.text, 1.25)
+			if math.Abs(tc.expWidth-w) > 0.0001 {
```
Comment (nikolaydubina) [inline render/render_test.go:26]: new lexical scope
```suggestion
			if w := textWidth(tc.text, 1.25); math.Abs(tc.expWidth-w) > 0.0001 {
```

**Pull request merged**